File: abstractions/java/lib/src/main/java/com/microsoft/kiota/authentication/AuthenticationProvider.java
package com.microsoft.kiota.authentication;

import com.microsoft.kiota.RequestInformation;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/** Authenticates the application request. */
public interface AuthenticationProvider {
    /**
     * Authenticates the application request.
     * @param request the request to authenticate.
     * @param additionalAuthenticationContext Additional authentication context to pass to the authentication library.
     * @return a CompletableFuture to await for the authentication to be completed.
     */
    @Nonnull
    CompletableFuture<Void> authenticateRequest(@Nonnull final RequestInformation request, @Nullable final Map<String, Object> additionalAuthenticationContext);
}

File: abstractions/php/src/Authentication/AuthenticationProvider.php
<?php
namespace Microsoft\Kiota\Abstractions\Authentication;

use Http\Promise\Promise;
use Microsoft\Kiota\Abstractions\RequestInformation;

/**
 * Interface AuthenticationProvider
 *
 * Authenticates the application request
 *
 * @package Microsoft\Kiota\Abstractions\Authentication
 * @copyright 2022 Microsoft Corporation
 * @license https://opensource.org/licenses/MIT MIT License
 * @link https://developer.microsoft.com/graph
 */
interface AuthenticationProvider {
    /**
     * @param RequestInformation $request
     * @return Promise
     */
    public function authenticateRequest(RequestInformation $request): Promise;
}
